### ISSUE TITLE
Touch ID not working

### ISSUE THREAD

**Issue by Mirumcer** [TablePlus/TablePlus#3018]
### Please answer the questions below, it helps us to keep track of the issue. Please do not ignore it or your issue will be closed.

1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81): 500

3. The steps to reproduce this issue: enable touch ID 
<img width="474" alt="image" src="https://github.com/TablePlus/TablePlus/assets/7575825/5f9e6fc6-eb7c-4b1d-9152-6322417c2a0b">
<img width="375" alt="image" src="https://github.com/TablePlus/TablePlus/assets/7575825/34b0bd70-5ce6-43cf-84c7-0ecd8bf60b28">

Noted: If the bug is related to data, please attach an example SQL data.

**Comment by ggedde** [comment]
Same for me. I get no Prompt to use Touch ID.
Also if I just try and use it nothing happens. I still have to use the passcode.

I have restarted TablePlus and also restarted my Macbook, but nothing.
I have gone in to the settings and disabled them and re-enabled them, still nothing.

Is there anything else I can check, I would really like to use this feature.

Thanks,

Version 6.2.1 (578)
MacOs 15.2 (24C101)

**Comment by huyphams** [comment]
Are you able to use the touch ID on your setup with other app @ggedde ?

**Comment by ggedde** [comment]
@huyphams Yes, Enpass, Phone Mirroring, and Passwords all ask for Touch ID and work. Enpass is a custom app while the other two are mac apps.

**Comment by traed** [comment]
Same issue here, also on build 578 and TouchID also works fine for other apps.

**Comment by huyphams** [comment]
I think the issue was introduced recently in macOS. I will add an error report to investigate the root cause. In the meantime, you can just close the popup and try again—it should work.

**Comment by traed** [comment]
What popup do you mean? We never get prompted for TouchID so there is no popup to close afaik.

**Comment by traed** [comment]
I don't believe this issue is about safe mode, it's about the option to lock the entire application with a passcode and TouchID. @huyphams

**Comment by huyphams** [comment]
I was trying to tell you that Touch ID will be prompted when Safe Mode is called, but somehow it failed, so only the Safe Mode popup appeared. If we close the Safe Mode popup and try the action again, it may work. At least on our end, where Touch ID rarely fails, I have only seen it fail once or twice in recent macOS versions. It may have something to do with macOS itself, which is why I said I need to add error reporting to identify the actual root cause.

**Comment by traed** [comment]
This issue is still not about unlocking safe mode with Touch ID, but rather using Touch ID to unlock the application itself when checking the "Enable Passcode" and "Unlock Passcode with Touch ID" options as shown in the original issue comment.

However, I found something out just now. If we check ALL three options "Enable Passcode", "Unlock Safe mode with Touch ID" and "Unlock Passcode with Touch ID" the Touch ID popup works when opening the application. If we have DON'T check "Unlock Safe Mode with Touch ID" but DO check "Enable Passcode" and "Unlock Passcode with Touch ID" the popup does not work when opening the application.

So it seems "Unlock Passcode with Touch ID" depends on the state of "Unlock Safe mode with Touch ID" which seems like a bug.

**Comment by huyphams** [comment]
Made sense @traed so I think it is the root cause, the app somehow requires both need to be enabled. It could be misconfigure the requirements. Should be check and fix in the up coming update.

**Comment by ggedde** [comment]
I too was trying it without safe mode. I have confirmed that enabling "Unlock Safe Mode with Touch ID" then closing and re-open did show the touch id for me.

However,  I then unchecked "Unlock Safe Mode with Touch ID" then closed the app and re-opened it and touch ID still shows up. So it seems to be working for me now, but yes this is a bug and should be resolved for those who never used or uses "Unlock Safe Mode with Touch ID".  Thanks!

**Comment by ggedde** [comment]
@huyphams Ahh, yes, I checked for updates earlier today and it said I was already up to date. Didn't realize I had auto updates enabled. So yeah, it could have started working because I got the latest updates.

Thanks!
